tests/test_utils.py

```diff
@@ -102,3 +104,153 @@ class MyObj:
+    assert SemanticVersion(major="1", minor="2", patch="3")
+    assert SemanticVersion(major=1, minor="2", patch="dev")
+    assert SemanticVersion(major=1, minor="2", patch="dev01")
+
+    assert SemanticVersion("1.1.dev")
+    assert SemanticVersion("1.1.dev1")
+
+    with pytest.raises(VersionSyntaxError):
+        SemanticVersion((1, 2))
+
+    with pytest.raises(VersionSyntaxError):
+        SemanticVersion(major=1, minor=2)
+
+    with pytest.raises(VersionSyntaxError):
+        SemanticVersion(major=1, minor=2, patch="dev.a")
+
+    with pytest.raises(VersionSyntaxError):
+        SemanticVersion(major=1, minor="dev", patch="dev1")
+
+    with pytest.raises(VersionSyntaxError):
+        SemanticVersion("1.1.deva")
+
+    with pytest.raises(VersionSyntaxError):
+        SemanticVersion("1.1")
+
+
+def test_semantic_version_methods():
+    ver = SemanticVersion("1.1.dev1")
+    assert ver.major == 1
+    assert ver.minor == 1
+    assert ver.patch == "dev1"
+
+    assert ver.as_string() == "1.1.dev1"
+    assert ver.as_tuple() == (1, 1, "dev1")
+    assert ver.as_dict() == {"major": 1, "minor": 1, "patch": "dev1"}
+    assert ver.as_list() == [1, 1, "dev1"]
+
+
+def test_semantic_version_comparison():
+    assert SemanticVersion("1.1.1") < SemanticVersion((1, 1, 2))
+    assert SemanticVersion("1.1.1") > SemanticVersion((1, 1, 0))
+
+    assert SemanticVersion("1.1.1") <= SemanticVersion((1, 1, 2))
+    assert SemanticVersion("1.1.1") >= SemanticVersion((1, 1, 0))
+
+    assert SemanticVersion("1.1.1") <= SemanticVersion((1, 1, 1))
+    assert SemanticVersion("1.1.1") >= SemanticVersion((1, 1, 1))
+
+    assert SemanticVersion("1.1.2") == SemanticVersion((1, 1, 2))
+    assert SemanticVersion("1.1.2") != SemanticVersion((1, 1, 3))
+
+    # checking major minors
+    assert SemanticVersion("1.1.1") < SemanticVersion((1, 2, 1))
+    assert SemanticVersion("1.1.1") > SemanticVersion((1, 0, 1))
+
+    assert SemanticVersion("1.1.1") < SemanticVersion((2, 1, 1))
+    assert SemanticVersion("1.1.1") > SemanticVersion((0, 1, 1))
+
+
+def test_semantic_version_comparison_dev():
+    assert SemanticVersion("1.1.1") < SemanticVersion((1, 1, "dev"))
+    assert SemanticVersion("1.1.1") < SemanticVersion((1, 1, "dev1"))
+    assert SemanticVersion((1, 1, "dev")) > SemanticVersion("1.1.1")
+    assert SemanticVersion((1, 1, "dev1")) > SemanticVersion("1.1.1")
+
+    assert SemanticVersion("1.1.dev1") == SemanticVersion((1, 1, "dev1"))
+    assert SemanticVersion("1.1.dev1") != SemanticVersion((1, 1, "dev0"))
+
+    assert SemanticVersion((1, 1, "dev1")) == SemanticVersion("1.1.dev1")
+    assert SemanticVersion((1, 1, "dev0")) != SemanticVersion("1.1.dev1")
+
+    # checking major minors
+    assert SemanticVersion((1, 2, "dev")) > SemanticVersion("1.1.1")
+    assert SemanticVersion((1, 0, "dev1")) < SemanticVersion("1.1.1")
+
+    with pytest.raises(ValueError, match="'dev' versions cannot be compared"):
```

Comment:
I was wondering if we should impose "devZ", being "Z" any positive integer. This way, we could always compare "dev" suffix versions.

What do you think, @germa89 @RobPasMue?

Reply:
In fact, we could just check against that regex honestly.

```py
import re
...

if "dev" in version and re.match("dev[0-9]+", version)
```

Reply:
That would get rid of the "magic number" for the length and force whatever Z to be a positive number

Reply:
Well... see PyFluent: https://github.com/pyansys/pyfluent/releases/tag/v0.12.dev14

Reply:
Why they don't just use a patch? xD

Reply:
I do not have the man power now to accomplish this. We can leave this ``'dev'`` thing for a later PR?

So I can close a PR in PyMAPDL.

Reply:
Yeah sure, let's keep track of it here: https://github.com/pyansys/pyansys-tools-versioning/issues/60#issuecomment-1376985422